e2e/tests/distribution_inspect.go

```diff
@@ -0,0 +1,201 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package tests
+
+import (
+	b64 "encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/gbytes"
+	"github.com/runfinch/common-tests/command"
+	"github.com/runfinch/common-tests/ffs"
+	"github.com/runfinch/common-tests/fnet"
+	"github.com/runfinch/common-tests/option"
+
+	registrytypes "github.com/docker/docker/api/types/registry"
+	"github.com/runfinch/finch-daemon/api/response"
+	"github.com/runfinch/finch-daemon/e2e/client"
+)
+
+// DistributionInspect tests `GET distribution/{name:.*}` API.
+func DistributionInspect(opt *option.Option) {
+	Describe("inspect a distribution", func() {
+		var (
+			uClient      *http.Client
+			version      string
+			registry     string
+			authImageTag string
+		)
+		BeforeEach(func() {
+			// create a custom client to use http over unix sockets
+			uClient = client.NewClient(GetDockerHostUrl())
+			// get the docker api version that will be tested
+			version = GetDockerApiVersion()
+		})
+
+		BeforeEach(func() {
+			filename := "htpasswd"
+			// The htpasswd is generated by
+			// `<subject> run --entrypoint htpasswd public.ecr.aws/docker/library/httpd:2 -Bbn testUser testPassword`.
+			// We don't want to generate it on the fly because:
+			// 1. Pulling the httpd image can take a long time, sometimes even more 10 seconds.
+			// 2. It's unlikely that we will have to update this in the future.
+			// 3. It's not the thing we want to validate by the functional tests. We only want the output produced by it.
+			// Same password/logic from runfinch/common-tests
+			htpasswd := "testUser:$2y$05$wE0sj3r9O9K9q7R0MXcfPuIerl/06L1IsxXkCuUr3QZ8lHWwicIdS"
```

Comment:
are we storing passord in plain text?

Reply:
It's just the password used to log into a local registry that's started as part of the the e2e tests. We need to put it in a particular file/directory so that clients can login to the registry server